public/app/features/dashboard-scene/serialization/sceneVariablesSetToVariables.test.ts

```diff
@@ -190,6 +190,131 @@ describe('sceneVariablesSetToVariables', () => {
     `);
   });
 
+  it('should handle Query variable when sceneVariablesSetToVariables should discard options', () => {
+    const variable = new QueryVariable({
+      name: 'test',
+      label: 'test-label',
+      description: 'test-desc',
+      value: ['selected-value'],
+      text: ['selected-value-text'],
+      datasource: { uid: 'fake-std', type: 'fake-std' },
+      query: 'query',
+      options: [
+        { label: 'test', value: 'test' },
+        { label: 'test1', value: 'test1' },
+        { label: 'test2', value: 'test2' },
+      ],
+      includeAll: true,
+      allValue: 'test-all',
+      isMulti: true,
+    });
+
+    const set = new SceneVariableSet({
+      variables: [variable],
+    });
+    const result = sceneVariablesSetToVariables(set);
+    expect(result).toHaveLength(1);
+    expect(result[0]).toMatchInlineSnapshot(`
+    {
+      "allValue": "test-all",
+      "current": {
```

Comment:
better to assert on what you care about than compare the full snapshot, snapshot tests are so painful in 99% of cases. Since this test cares about options, check if options are there or not